.claude/hooks/pre_tool_use_protect_precommit.py

```
@@ -0,0 +1,189 @@
+def is_protected_config_file(file_path) -> bool:
+    """Check if a file path matches protected configuration patterns."""
+    if not file_path:
+        return False
+    path_str = str(file_path)
+    for pattern in PROTECTED_CONFIG_FILES:
+        if re.search(pattern, path_str, re.IGNORECASE):
+            return True
+    return False
+
+
+def check_config_bash_command(command):
```

Ruff (0.12.2)

Missing return type annotation for public function `check_config_bash_command`

(ANN201)